connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1090,6 +1090,39 @@ public void putConnectorConfig(final String connName, final Map<String, String>
         );
     }
 
+    @Override
+    public void stopConnector(final String connName, final Callback<Void> callback) {
+        log.trace("Submitting request to transition connector {} to STOPPED state", connName);
+
+        addRequest(
+                () -> {
+                    refreshConfigSnapshot(workerSyncTimeoutMs);
+                    if (!configState.contains(connName))
+                        throw new NotFoundException("Unknown connector " + connName);
+
+                    // We only allow the leader to handle this request since it involves writing task configs to the config topic
+                    if (!isLeader()) {
+                        callback.onCompletion(new NotLeaderException("Only the leader can transition connectors to the STOPPED state.", leaderUrl()), null);
+                        return null;
+                    }
+
+                    // TODO: We may want to add a new ConfigBackingStore method for stopping a connector so that
```

Comment:
I'm worried that if we create a ticket, somebody might jump on that and make the change now, even though it's not really necessary yet. If it's too bothersome as a TODO I could just remove it altogether?

Reply:
Usually I prefer TODO tags to be associated with a task. What's the drawback if someone was to make this change now?

Reply:
Currently, the `KafkaConfigBackingStore` uses a transactional producer only when exactly once support is enabled on the worker. Maybe Chris' concern was that currently we'll only be able to do the tasks config + target state write atomically when exactly once support is enabled on the worker? 🤔 

Although that is already the case today for other operations that require multiple writes to the config topic like `putTaskConfigs` and `removeConnectorConfig`, so I'm not sure whether or not that was his concern.

Reply:
I guess this wasn't really a `TODO` so much as a `MAYBE-DO`. I've removed it since on second thought it's a `DON'T-DO-FOR-NOW` as there is little if any benefit from that change, except possibly blocking the herder thread for less time if we're having trouble writing to the config topic.